src/quorum/datadriven_test.rs

```diff
@@ -123,7 +123,76 @@ fn test_quorum(data: &TestData) -> String {
                 }
                 buf.push_str(&format!("{}\n", idx.0));
             } else {
-                // TODO(accelsao): port majority commit
+                use_group_commit = false;
+
+                let idx = c.committed_index(use_group_commit, &l);
+                buf.push_str(&c.describe(&l));
+
+                // Joining a majority with the empty majority should give same result.
+                let a_idx =
+                    JointConfig::new_joint_from_majorities(c.clone(), MajorityConfig::default())
+                        .committed_index(use_group_commit, &l);
+                if a_idx != idx {
+                    buf.push_str(&format!("{} <-- via zero-joint quorum\n", a_idx.0));
+                }
+
+                // Joining a majority with itself should give same result.
+                let a_idx = JointConfig::new_joint_from_majorities(c.clone(), c.clone())
+                    .committed_index(use_group_commit, &l);
+                if a_idx != idx {
+                    buf.push_str(&format!("{} <-- via self-joint quorum\n", a_idx.0));
+                }
+
+                // test overlaying
```

Comment:
Better also port the comment here.
`If the committed index was definitely above the currently inspected idx, the result shouldn't change if we lower it further.`